dvyukov covfilter minor cleanups

dvyukov opened the pull request:
- executor: minor coverage filter cleanup
- syz-manager: minor coverage filter cleanup

dvyukov commented:
@Bins94 please take a look at these changes

Bins94 commented:
[888af9b](https://github.com/google/syzkaller/commit/888af9bb7ef25b201fa033a62af97f1ca0f7db04)

I couldn't see there is any negative affect here.
In this case:

>  ["files": "bpf"]

 It may filter comparison constants of syscall argument parsing before executing subsystem code. But I believe syzkaller system call descriptions are powerful enough to cover these codes. I always enable all syscalls and specify target only, it works well also. Also, if it's necessary, users can add kernel functions into the coverage filter, it requires users to have enough knowledge to enable function rightly.

dvyukov commented:
> It may filter comparison constants of syscall argument parsing before executing subsystem code.

I would assume that if there are any interesting comparisons happening in some code, that code needs to be included into the coverage filter. If there are comparisons, most likely there are also branches on the results of these comparisons, so we very much want to use that coverage as the signal as well.
If we use comparison operands to cover new code paths, we very much want to persist these paths in the corpus. But if this code is not in the coverage filter, we will throw these new interesting inputs away.
Some corner cases are possible, of course, where comparisons are interesting but code coverage is somehow not interesting. But I believe these are very narrow corner cases and it's reasonable to include these small pieces of code into the coverage filter.

dvyukov commented:
Added commit that implements showing filtered coverage percent:
```
coverage | 55058
filtered coverage | 7945 / 16606 (47%)
```
https://github.com/google/syzkaller/pull/2318/commits/56bc5b6f9224210505b335a4f8c921778407fc69

Also dropped commit that filters coverage in executor.

Bins94 commented:
> > It may filter comparison constants of syscall argument parsing before executing subsystem code.
> 
> I would assume that if there are any interesting comparisons happening in some code, that code needs to be included into the coverage filter. If there are comparisons, most likely there are also branches on the results of these comparisons, so we very much want to use that coverage as the signal as well.
> If we use comparison operands to cover new code paths, we very much want to persist these paths in the corpus. But if this code is not in the coverage filter, we will throw these new interesting inputs away.
> Some corner cases are possible, of course, where comparisons are interesting but code coverage is somehow not interesting. But I believe these are very narrow corner cases and it's reasonable to include these small pieces of code into the coverage filter.

Agree, and I had run the code. It looks good to me. It seems to have a slight improvement, and I am still testing.

dvyukov commented:
> It seems to have a slight improvement, and I am still testing.

Note: we are collecting a bit more signal now:
https://github.com/google/syzkaller/pull/2318/commits/6188105a760e0cf5ea9985fb8621c8eb06715bea
It may explain some increase in signal.

Bins94 commented:
> Note: we are collecting a bit more signal now:
> [6188105](https://github.com/google/syzkaller/commit/6188105a760e0cf5ea9985fb8621c8eb06715bea)
> It may explain some increase in signal.

I know that, I pulled your [branch](https://github.com/dvyukov/syzkaller/tree/dvyukov-covfilter-minor-cleanups), read every commit change, and run the code.
Everything is good to me. Especially collecting both incoming and outgoing edges, I didn't realize this problem before.

dvyukov commented:
Thanks for testing!
I will merge this now then. And then we can do any additional tweaks/improvements in separate smaller patches if necessary.

Bins94 commented:
Happy to see the coverage filter become easier to use and more efficient! :)
